docs/source/starter/style_guide.rst

```diff
@@ -24,41 +26,60 @@ In Lightning we differentiate between a system and a model.
 
 A model is something like a resnet18, RNN, etc.
 
-A system defines how a collection of models interact with each other. Examples of this are:
+A system defines how a collection of models interact with each other with user-defined training/evaluation logic. Examples of this are:
 
 * GANs
 * Seq2Seq
 * BERT
-* etc
+* etc...
 
 A LightningModule can define both a system and a model.
 
-Here's a LightningModule that defines a model:
+Here's a LightningModule that defines a model. Although, we do not recommend to define a model like in the example.
 
 .. testcode::
 
     class LitModel(LightningModule):
-        def __init__(self, num_layers: int = 3):
+        def __init__(self):
             super().__init__()
             self.layer_1 = nn.Linear()
             self.layer_2 = nn.Linear()
             self.layer_3 = nn.Linear()
 
-Here's a LightningModule that defines a system:
+Here's a LightningModule that defines a system. This structure is what we recommend as a best practice.
```

Comment:
This should come first in the docs since it's the recommended path to follow 

Somewhere here we should emphasize that this distinction improves modularity, which has a slew of benefits (testability, reduced dependencies, easier to refactor)